l10n_it_fatturapa/bindings/fatturapa.py

```diff
@@ -0,0 +1,149 @@
+# -*- coding: utf-8 -*-
+
+import logging
+import pyxb.binding
+from odoo.modules.module import get_module_resource
+
+from lxml import etree
+from binding import *  # noqa: F403
+
+_logger = logging.getLogger(__name__)
+_logger.setLevel(logging.DEBUG)
+
+XSD_SCHEMA = 'Schema_del_file_xml_FatturaPA_versione_1.2.1.xsd'
+
+_xsd_schema = get_module_resource('l10n_it_fatturapa', 'bindings', 'xsd',
+                                  XSD_SCHEMA)
+_root = etree.parse(_xsd_schema)
+
+_CreateFromDocument = CreateFromDocument  # noqa: F405
+
+date_types = {}
+datetime_types = {}
+string_types = {}
+
+
+def get_parent_element(e):
+    for ancestor in e.iterancestors():
+        if 'name' in ancestor.attrib:
+            return ancestor
+
+
+def get_type_query(e):
+    return "//*[@type='%s']" % e.attrib['name']
+
+
+def collect_element(target, element, parent=None):
+    if parent is None:
+        parent = get_parent_element(element)
+
+    path = '//%s/%s' % (parent.attrib['name'], element.attrib['name'])
+    mandatory = element.attrib.get('minOccurs') != '0'
+    if path not in target:
+        target[path] = mandatory
+    else:
+        assert target[path] == mandatory, 'Element already present with ' \
+                                          'different minOccurs valus'
```

Comment:
```suggestion
                                          'different minOccurs value'
```
Si potrebbe anche aggiungere il `path` (e volendo anche il `text`) per uniformarlo agli altri messaggi di warning e poter capire dai log che sta succedendo

Reply:
ma è un assert, cmq dimmi cosa scrivo, ce lo metto, non dovrebbe capitare

Reply:
```
        if target[path] != mandatory:
            msg = "Element having path %s already present with a different minOccurs value" % path
            _logger.warn(msg)
```

Reply:
no deve proprio scoppiare se capita :)

Reply:
Va bene, secondo me è utile indicare il path di cosa sta fallendo nei log per capire cosa sta succedendo e dove.
Se usarlo per sollevare eccezione o con un assert decidi tu

Reply:
si ora riproduco aggiungendo un doppione nell'xsd e vedo come viene meglio l'errore

Reply:
credo una cosa simile sia accettabile:
`Element //DatiTrasporto/Descrizione is already present with different minOccurs value`